Add Microsoft OpenJDK Buildpack to Paketo Buildpacks

dmikusa-pivotal opened the pull request:
## Summary

Microsoft now has a build of OpenJDK. This request will add an official buildpack to supply Microsoft's OpenJDK.

## Use Cases

This is going to be the recommended OpenJDK to use when deploying Java apps to Azure.

## Checklist
<!-- Please confirm the following -->
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

ekcasey commented:
@dmikusa-pivotal Let's make this a Java subteam RFC instead of a top-level RFC?

dmikusa-pivotal commented:
I'm fine with that. 

@ryanmoran suggested this route. Does moving this to a Java subteam RFC make sense to you as well? 

We can discuss on Tue in the working group, if that would be preferred. I assume the same will go for #77 as well.

Thanks

ekcasey commented on text/0000-microsoft-jdk.md:
```diff
@@ -0,0 +1,39 @@
+
+## Motivation
+
+We would like to move the Microsoft OpenJDK Buildpack into the Paketo Community so that the community can use it to build Java applications, in particular on Microsoft Azure. This will also give the buildpack maintainer the ability to leverage common Paketo tooling to keep the buildpack dependencies updated and release the buildpack.
```
I think this can go directly into `paketo` and skip `paketo-community`. This buildpack is nearly identical other JVM buildpacks and therefore the risk here seems very low and this shouldn't need to incubate in community.

dmikusa-pivotal replied:
Done.

dmikusa-pivotal commented:
I have moved this RFC into the Java subteam.

I have also moved this so it will go directly into Paketo Buildpacks. This is because the buildpack is already using existing tooling and is ready to move directly.

I have also clarified that maintainership will be done by the Java team as a team needs to own the buildpack and individuals can be added to the team using existing governance processes.

dmikusa-pivotal commented on text/0000-microsoft-jdk.md:
```diff
@@ -0,0 +1,39 @@
+
+## Summary
+
+A [Microsoft OpenJDK Buildpack](https://github.com/eddumelendez/microsoft-openjdk) for supplying Microsoft's distribution of OpenJDK has been created by [@eddumelendez][@eddumelendez]. It is built using common tooling (libpak and lijvm) used by the Paketo Java sub-team and should be adopted into the Paketo Community Github Org.
```
Fixed typo. Thanks.
